### PR TITLE
feat: groups follow api

### PR DESCRIPTION
**Pull request by MeghalBisht**
as per meeting notes

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The unit tests pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

- [ ] All the functions created/modified in this PR contain relevant docstrings.

### CODE REVIEW

**Review by codedsun — APPROVED** [review]

**Comment by sachinchauhan2889** [comment]
Meghal now all end points are working. You can check them by postman. Permissions Part is not checked by me.

**Comment by sachinchauhan2889** [comment]
pushed dropdown changes here by mistake. I have reverted. :smile:

**Review by ranjitkshah — DISMISSED** [review]

File: app/api/user_follow_groups.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+    List User Followed Groups
+    """
+
+    def query(self, view_kwargs):
+        query_ = UserFollowGroup.query
+
+        if view_kwargs.get('user_id') is not None:
+            user = safe_query_kwargs(User, view_kwargs, 'user_id')
+            if user != current_user or not is_logged_in():
+                raise ForbiddenError({'pointer': 'user_id'})
```
Comment (maze-runnar) [inline app/api/user_follow_groups.py:64]: should we use has_oganizer_acess here too?

**Review by maze-runnar — COMMENTED** [review]
lgtm

File: app/api/user_follow_groups.py (hunk 1 above)
Comment (MeghalBisht) [reply app/api/user_follow_groups.py:64]: I am not sure, @sachinchauhan2889 what do you say?
In the meet it was discussed, normal users can also follow group.

**Review by sachinchauhan2889 — APPROVED** [review]
LGTM.

Will discuss permissions, testing, api docs for this feature in upcoming meeting.

**Comment by ranjitkshah** [comment]
> LGTM.
> 
> Will discuss permissions, testing, api docs for this feature in upcoming meeting.

Follow up in next pr's

**Review by ranjitkshah — APPROVED** [review]

**Pull request merged**